tensorboard/components/tf_storage/storage.ts

```diff
@@ -51,82 +51,155 @@ export let TAB = '__tab__';
  * it is NOT safe to change the disambiguator string without find+replace
  * across the codebase.
  */
-export let DISAMBIGUATOR = 'disambiguator';
-
-/**
- * Return a string stored in URI or localStorage.
- * Undefined if not found.
- */
-export function getString(key: string, useLocalStorage: boolean): string {
-  if (useLocalStorage) {
-    return window.localStorage.getItem(key);
-  } else {
-    return _componentToDict(_readComponent())[key];
-  }
+export const DISAMBIGUATOR = 'disambiguator';
+
+export const {
```

Comment:
Cool, I haven't seen this syntax before. It seems like a nice use of polymorphism. Where's the name of the exported interface specified (How are they used?)?

Reply:
No polymorphism here. In the code
```ts
const stuff = {one: 1, two: 2, three: 3};
const {one} = stuff;  // A
const {two: dos} = stuff;  // B
export const {three: tres} = stuff;  // C
```
the last three lines are equivalent to
```ts
const one = stuff.one;
const dos = stuff.two;
export const tres = stuff.three;
```

The names of the exported interface are `getString`, `setString`, `getStringInitializer`, and `getStringObserver`.

These are used when someone writes `import {getStringInitializer, getStringObserver} from '../tf-storage/storage.js'` (or `import * as storage from '../tf-storage/storage.js'`, etc.).

Relevant MDN page: [Destructuring assignment &sect; Assigning to new variable names](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Assigning_to_new_variable_names).

Reply:
Got it. Thanks! I wasn't aware of that expansion. I was referring to the parametric polymorphism of `makeBindings` - we can templatize it for any of the types.

Reply:
Ah—yep, that is indeed parameterly polymorphic. (Sorry, I'd assumed that you meant object polymorphism.)